packages/components/src/DataTable/Body.tsx

```diff
@@ -25,34 +28,57 @@ export function Body<T extends object>({
     [onRowClick],
   );
 
-  return table.getRowModel().rows.length ? (
-    <tbody>
-      {table.getRowModel().rows.map(row => {
-        return (
-          <tr
-            key={row.id}
-            onClick={handleRowClick(row)}
-            className={bodyRowClasses}
-          >
-            {row.getVisibleCells().map(cell => {
+  const loaderRows = table.getState().pagination.pageSize;
+  const loaderColumns = table.getAllColumns().length;
+  return (
+    <>
+      {!loading ? (
```

Comment:
I think it would be good to do an early return instead of adding another ternary to this `return`

It can be as simple as 
```
  if (loading) {
    return (
      <tbody>
        {Array.from(Array(loaderRows).keys()).map(row => (
          <tr key={row}>
            {Array.from(Array(loaderColumns).keys()).map(arr => (
              <td key={arr}>
                <Glimmer />
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    );
  }
```

It you want to go further, this could be extracted to its own component, something like `DataTableLoadingState`, but it's up to you, I see that in this file there's a pattern of doing ternaries and this would be an improvement, but not necessary for now.

Reply:
Good point, I agree! Changed the pattern let me know what you think.